acoular/fprocess.py

```diff
@@ -0,0 +1,367 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) Acoular Development Team.
+# ------------------------------------------------------------------------------
+"""Implements blockwise processing methods in the frequency domain.
+
+.. autosummary::
+    :toctree: generated/
+
+    RFFT
+    IRFFT
+    AutoPowerSpectra
+    CrossPowerSpectra
+    FFTSpectra
+"""
+
+import multiprocessing
+from warnings import warn
+
+import numpy as np
+from scipy import fft
+from traits.api import Bool, CArray, Either, Instance, Int, Property, Trait, cached_property
+
+from .base import SamplesGenerator, SpectraGenerator, SpectraOut, TimeOut
+from .fastFuncs import calcCSM
+from .internal import digest
+from .spectra import BaseSpectra
+from .tools.utils import SamplesBuffer
+
+CPU_COUNT = multiprocessing.cpu_count()
+
+
+class RFFT(BaseSpectra, SpectraOut):
+    """Provides the one-sided Fast Fourier Transform (FFT) for real-valued multichannel time data.
+
+    The FFT is calculated block-wise, i.e. the input data is divided into blocks of length
+    :attr:`block_size` and the FFT is calculated for each block. Optionally, a window function
+    can be applied to the data before the FFT calculation via the :attr:`window` attribute.
+    """
+
+    #: Data source; :class:`~acoular.base.SamplesGenerator` or derived object.
+    source = Instance(SamplesGenerator)
+
+    #: Number of workers to use for the FFT calculation
+    workers = Int(CPU_COUNT, desc='number of workers to use')
```

Comment:
` workers = Int(-1, desc='number of workers to use')`
would also work, because then the output of `os.cpu_count()` ist used.

Reply:
See my previous comment.